app/scripts/directives/fromFile.js

```diff
@@ -137,30 +153,41 @@ angular.module("openshiftConsole")
             resourceCheckPromises.push(checkIfExists(item));
           });
 
-          $q.all(resourceCheckPromises).then(function() {
-            if ($scope.errorOccured) {
-              return;
-            }
-            // If resource is Template and it doesn't exist in the project
-            if ($scope.createResources.length === 1 && $scope.resourceList[0].kind === "Template") {
-              openTemplateProcessModal();
-            // Else if any resources already exist
-            } else if (!_.isEmpty($scope.updateResources)) {
-              $scope.updateTemplate = $scope.updateResources.length === 1 && $scope.updateResources[0].kind === "Template";
-              if ($scope.updateTemplate) {
+          createProjectIfNecessary().then(function(project) {
+            $scope.input.selectedProject = project;
+            $q.all(resourceCheckPromises).then(function() {
+              if ($scope.errorOccured) {
```

Comment:
Actually looks like it's misspelled here in the variable name too but that's not new